zjit/src/codegen.rs

```diff
@@ -154,6 +154,19 @@ fn gen_iseq_entry_point_body(cb: &mut CodeBlock, iseq: IseqPtr) -> *const u8 {
     start_ptr.map(|start_ptr| start_ptr.raw_ptr(cb)).unwrap_or(std::ptr::null())
 }
 
+fn register_with_perf(iseq_name: String, start_ptr: usize, code_size: usize) {
+    use std::io::Write;
+    let perf_map = format!("/tmp/perf-{}.map", std::process::id());
+    let Ok(mut file) = std::fs::OpenOptions::new().create(true).append(true).open(&perf_map) else {
```

Comment:
I think we had found doing something like this was slow for YJIT. Just like YJIT, could you create the file when the option is parsed, keep the fd, and just write to the fd here?

Reply:
Oh sorry, my bad. I mixed up disasm and perf. Never mind!